File: tests/fixtures/vulnerable/sqli.py
# FIXTURE: SQL injection, Python.
import sqlite3

def get_user_concat(cursor, user_id):
    # EXPECT sql-injection
    cursor.execute("SELECT * FROM users WHERE id = " + user_id)

def get_user_fstring(cursor, name):
    # EXPECT sql-injection
    cursor.execute(f"SELECT id, email FROM users WHERE name = '{name}'")

def get_user_percent(cursor, email):
    # EXPECT sql-injection
    cursor.execute("SELECT * FROM users WHERE email = '%s'" % email)

File: tests/fixtures/vulnerable/sqli.ts
// FIXTURE: SQL injection, TypeScript.
interface Conn { execute(sql: string): Promise<unknown>; }

export async function findOrder(conn: Conn, orderId: string): Promise<unknown> {
  // EXPECT sql-injection
  return conn.execute(`SELECT * FROM orders WHERE order_id = '${orderId}'`);
}
